### PR TITLE
feat(SidePanel): new side panel tab look-and-feel

### PR DESCRIPTION
**Pull request by jbocce**
### Context

With more and taller (side) panels projected in the future, the side panel tabs at the top to switch between the various panels should be streamlined to allow more vertical space and presented (all) on screen so that one gets a sense of which panels are available.

Thanks to @dan-rukas, there are several looks for the tabs based on the number of panels on a particular side.

For one panel a text label continues to be used. Note that the label should be centered horizontally relative to the entire side panel.

image

For two tabs, the tabs are 68px wide and 28px tall and look as below. The two tabs must also be centered horizontally relative to the entire panel width.

image

With three or more tabs, the tabs are 40px wide and 28px tall. Again the tabs are centered horizontally relative to the entire panel, however if centering them in that manner would overlap the close button, the tabs are positioned directly to the left of the left panel close button OR directly to the right of the right panel close button.

image

### Changes & Results

The design was implemented.

The former SidePanel component was renamed LegacySidePanel.

The dependency on services for the SidePanel was removed and a new SidePanelWithServices was introduced.

### Testing

Add various panels to either side. The new design should be applied accordingly.

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [x] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] OS: Windows 11
- [x] Node version: 16.14.0
- [x] Browser: Chrome 116.0.5845.188
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: platform/ui/src/components/SidePanel/SidePanel.tsx [hunk 1]
```diff
@@ -52,95 +51,119 @@ const classesMap = {
...
+  const openStatus = panelOpen ? 'open' : 'closed';
+  const style = Object.assign({}, styleMap[openStatus][side], baseStyle);
+
+  const ActiveComponent = tabs[activeTabIndex]?.content;
 
   const updatePanelOpen = useCallback((panelOpen: boolean) => {
     setPanelOpen(panelOpen);
     if (panelOpen) {
-      setHasBeenOpened(true);
+      onOpened?.();
```
Comment (sedghi) [inline platform/ui/src/components/SidePanel/SidePanel.tsx:147]: I'm not sure if `onOpened` or `onOpen`, I feel like the second one match the rest of codebase better

**Review by sedghi — COMMENTED** [review]
See my only comment, looks great, thanks. 

Let's wait to see what Dan come up with the open/close Icon

**Comment by sedghi** [comment]
I see some strange behaviour, which looks like an instant "open" of the side panel on the left, as if someone clicking open very soon. It is a changed behaviour as previously we had it "start" from the open state 

https://github.com/OHIF/Viewers/assets/7490180/3068bbb9-3cdf-4ffd-9a02-c8d8e6a47765

**Comment by jbocce** [comment]
> I see some strange behaviour, which looks like an instant "open" of the side panel on the left, as if someone clicking open very soon. It is a changed behaviour as previously we had it "start" from the open state
> 
>  CleanShot.2023-09-18.at.14.37.38.mp4

Great catch. I think I fixed it. Please have a look.

**Comment by sedghi** [comment]
I feel like this icon is much huger than the right side icons

CleanShot 2023-09-22 at 12 40 16
http://localhost:3000/tmtv?StudyInstanceUIDs=1.3.6.1.4.1.14519.5.2.1.7009.2403.334240657131972136850343327463

**Pull request merged**